gemrb/core/Strings/UTF8Comparison.cpp

```diff
@@ -84,9 +87,14 @@ bool UTF8_stricmp(const char *a, const char *b) {
 				}
 			};
 
-			uint8_t widthA = getWidth(iA), widthB = getWidth(iB);
-
+#ifdef __APPLE__
```

Comment:
we should probably add a cmake detection for `C.UTF-8`... then again, why not just always do it this way? this should work for everything, no?

Reply:
You mean we check for `C.UTF-8` in some cmake step, and what is currently `#ifdef __APPPLE__` turns into `#ifndef HAVE_C_UTF8`?

Reply:
yes, but im also asking why not just drop the `#ifdef` and just do it the one way that works for everything.

Reply:
Simply because I've started with `C.UTF-8` as it sounds to have a wider compatibility, language-wise. If it works for the cases we have, we can give it a try, sure.

Reply:
well if they aren't equivalent (for our purposes), then we have a problem anyway. But all this is doing is creating a `C` locale, then setting `LC_CTYPE` to `UTF-8`, which is the same things as `C.UTF-8` as far as I can tell.

Reply:
Let's do it like this for now.